server/src/test/java/org/elasticsearch/search/fetch/subphase/FetchSourceContextTests.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.fetch.subphase;
+
+import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class FetchSourceContextTests extends AbstractSerializingTestCase<FetchSourceContext> {
+    @Override
+    protected FetchSourceContext doParseInstance(XContentParser parser) throws IOException {
+        return FetchSourceContext.fromXContent(parser);
```

Comment:
When I wrote some tests that recently had the NPE on this sort of thing I would do:
```
        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
       FetchSourceContext result = FetchSourceContext.fromXContent(parser);
       ensureExpectedToken(XContentParser.Token.END_OBJECT, parser.nextToken(), parser);
       return parser;
```

That way I don't have to modify the production code for the test. I think it's *probably* the right thing to do here too.

Reply:
if fetchSource equals false, the XContent is not object, nor json, it is a value of false.

Reply:
Indeed it does!